src/block/ChiseledBookshelf.php

```diff
@@ -48,11 +48,30 @@ class ChiseledBookshelf extends Opaque{
 	 */
 	private array $slots = [];
 
+	private ?ChiseledBookshelfSlot $lastInteractedSlot = null;
+
 	protected function describeBlockOnlyState(RuntimeDataDescriber $w) : void{
 		$w->horizontalFacing($this->facing);
 		$w->enumSet($this->slots, ChiseledBookshelfSlot::cases());
 	}
 
+	public function readStateFromWorld() : Block{
+		$tile = $this->position->getWorld()->getTile($this->position);
+		if($tile instanceof TileChiseledBookshelf){
+			$this->lastInteractedSlot = $tile->getLastInteractedSlot();
+		}
```

Comment:
This should have an `else` to set the property to `null` if the tile is missing

Reply:
This property is initialized to null by default. It isn't enough?

Reply:
On closer examination it appears some other blocks do the same thing you've done. Probably ought to be fixed, but that's for another time.

In core code this is only called in `getBlockAt()`, but there's no telling what plugins might do. There's no guarantee that the property is `null`.